c7n/resources/emr.py

```diff
@@ -284,3 +284,73 @@ def query(self):
             value = [self.value]
 
         return {'Name': self.key, 'Values': value}
+
+
+@resources.register('emr-block-public-access-configuration')
+class EMRBlockPublicAccessConfiguration(QueryResourceManager):
+    """Resource manager for EMR block public access configuration
+    """
+
+    class resource_type(TypeInfo):
+        service = 'emr'
+        enum_spec = ('get_block_public_access_configuration', None, None)
+        date = "BlockPublicAccessConfigurationMetadata.CreationDateTime"
+        name = id = 'BlockPublicAccessConfigurationMetadata.CreatedByArn'
+        arn_type = 'emr'
+        permission_prefix = 'elasticmapreduce'
+
+    def augment(self, resource):
+        config = resource
+        config.pop('ResponseMetadata')
+        config = [config]
+        return config
+
+
+@EMRBlockPublicAccessConfiguration.action_registry.register('put-block-public-access-configuration')
```

Comment:
Is there a way to know what an existing block-public-access-configuration in the region is configured for before updating? e.g. something to the effect of
```
filters:
  - not:
    - type: BlockPublicAccessConfiguration
       BlockPublicSecurityGroupRules: True   
       PermittedPublicSecurityGroupRuleRanges:
                - MinRange: 22
                   MaxRange: 22
                - MinRange: 23
                   MaxRange: 23
```

Reply:
Now I think this should be doable using the emr-block-public-access-configuration filter before doing the put-emr-block-public-access-configuration action.

Reply:
also use set over put for consistency as a prefix